modules/ROOT/pages/test-recorder.adoc

```diff
@@ -5,10 +5,41 @@ endif::[]
 
 The test recorder enables you to record a processing flow and then configure a unit test based on the captured event.
 
-The test recorder is an open beta feature that is compatible with Anypoint Studio 7.x.x and later and MUnit 2.2.1 and later.
+The test recorder captures real data as it goes through your application in Studio. By capturing the flow execution, MUnit can automatically generate an MUnit test so that you can configure the necessary mocks and assertions for your test. See xref:test-recorder-examples.adoc[Recording a Test from an Exchange Example] for a full working example.
+
+[NOTE]
+The test recorder is an open beta feature.
+
+== Compatibility
+
+The following table describes the required product versions to use the test recorder. These versions are for the beta feature and are likely to change for the GA (general availability) release.
+
+[%header%autowidth.spread,cols="a,a"]
+|===
+| Product | Version
+| Anypoint Studio
+| 7.4.0 and later.
+
+| MUnit
+| 2.2.4
+
+| Mule runtime engine
+| Patched version of 4.2.2. +
+See xref:runtime-patching.adoc[Patching the Embedded Mule Runtime Engine Version] to learn how to apply this patched version to your project.
+
+|===
 
 == Limitations
 
-== See Also
+The test recorder has the following limitations for creating MUnit tests:
+
+* You cannot create tests for flows that have mule errors either raised inside the flow or already inside the incoming event. Not even if they are handled by an *on error continue* error handler.
+* When the recorded flow execution ends, the application is killed. This causes the result of the execution to never reach its destination. For example, an HTTP listener will not respond to the incoming request. This does not mean that the recording failed.
+* The test recorded does not support creating tests for anything that uses parallelism, random data, or time dependent information (for example, timestamps).
```

Comment:
The test recorder, not the test recorded. And it is not that it is not supported, but rather that if configured incorrectly (e.g. try to assert a timestamp), then the test will always fail (e.g. the timestamp will always change). This is not an issue of the recorder per se, but rather illogic testing, regardless of the tool.

Reply:
If it's not a limitation of the feature, I wouldn't suggest adding it here. Although I see why it would be useful information for new users. 
I'd say:

> You should not write assertions  for anything that uses parallelism, random data, or time-dependent information (e.g. timestamps). Because those values will change in every execution, your assertions will fail every time your test runs.


Or something like that 🤔

Reply:
Exactly my point. The idea of the recorder is to bring in new users (as well as helping existing ones), and the tool will not stop you from doing these things, so putting it here helps fill that gap. What I would add to your suggestion is that both spies and assertions will fail in all these cases (spies are just assertions in the middle of the execution), but mocks will only fail with parallelism, and actually mocking random data or time-dependent information is actually something that would help make the execution more consistent and therefore easier to test. Not sure how to put all that mess into a coherent sentence, though. Maybe something like this?

> You should not configure spies or assertions for anything that uses parallelism, random data, or time-dependent information (for example, timestamps), because those values will change in every execution, and those actions will fail every time your test runs. For similar reasons, you should not mock anything that uses parallelism, because there will be a mix of real and mocked data that will break anything that follows.

Reply:
I'd go with an imperative voice for this sentence:

> * Don't configure Spy or Assert processors to assert values for random data, time-dependent information (such as timestamps), or values resulting from parallel processes. Because those values change in every execution, your validations will fail every time your test runs. 
>
> * Don't mock values resulting from parallel processes because the mixture of real and mocked data will compromise the execution of the processors that follow in your test.

Reply:
Looks great!